### PR TITLE
Add missing flags for the `astro check` command

### LINKED ISSUE #7542: Astro check command missing flags
### 📚 Subject area/topic

astro cli commands

### 📋 Page(s) affected (or suggested, for new content)

https://docs.astro.build/en/reference/cli-reference/#astro-check

### 📋 Description of content that is out-of-date or incorrect

the only flag listed on the docs website for the astro check command is `--watch`
There are it turns out other flags that can be supplied such as `--minimumSeverity` or `--help`

### 🖥️ Reproduction in StackBlitz (if reporting incorrect content or code samples)

_No response_

### PR DESCRIPTION
**Pull request by lifeisfakenews**
#### Description (required)

Added the flags that are missing from the current documentation for the `astro check` cli command
Currently, only the `--watch` flag is documented, however there are several other options available for the command (`--minimumSeverity` or `--preserveWatchOutput` for example)

#### Related issues & labels (optional)

- Closes #7542

### CODE REVIEW

File: src/content/docs/en/reference/cli-reference.mdx [hunk 1]
```diff
@@ -226,6 +226,34 @@ Use these flags to customize the behavior of the command.
...
+
+Default: `hint`
+
+Minimum severity to output.
+
+For example, choosing `warning` will show errors and warnings, but not hints.
+
+#### `--preserveWatchOutput`
+
+If provided, the output will not be cleared between checks when in watch mode.
```
Comment (at-the-vr) [inline src/content/docs/en/reference/cli-reference.mdx:255]:
```suggestion
#### `--preserveWatchOutput`

Default: `false`

If set to false, the output will not be cleared between checks when in watch mode.
```

**Review by at-the-vr — COMMENTED** [review]
Great PR 🙌 Got 1 suggestion to checkout

File: src/content/docs/en/reference/cli-reference.mdx [hunk 2]
```diff
@@ -226,6 +226,36 @@ Use these flags to customize the behavior of the command.
 
 The command will watch for any changes in your project, and will report any errors.
 
+#### `--root <path-to-dir>`
+
+Specify a different root directory to check. Uses the current working directory by default.
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:231]:
```suggestion
Specifies a different root directory to check. Uses the current working directory by default.
```

File: src/content/docs/en/reference/cli-reference.mdx [hunk 3]
```diff
@@ -226,6 +226,36 @@ Use these flags to customize the behavior of the command.
...
+
+#### `--tsconfig <path-to-file>`
+
+Specify a `tsconfig.json` or `jsconfig.json` file to use manually. If not provided, Astro will attempt to find a config, or infer the project's config automatically.
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:235]:
```suggestion
Specifies a `tsconfig.json` or `jsconfig.json` file to use manually. If not provided, Astro will attempt to find a config, or infer the project's config automatically.
```

File: src/content/docs/en/reference/cli-reference.mdx [hunk 4]
```diff
@@ -226,6 +226,36 @@ Use these flags to customize the behavior of the command.
...
+
+#### `--minimumFailingSeverity <error|warning|hint>`
+
+Default: `error`
+
+Minimum severity needed to exit with an error code.
+
+For example, chooisng `warning` will cause the command to exit with an error if any warnings are detected.
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:243]:
```suggestion
Specifies the minimum severity level needed to exit with an error code. Defaults to `error`.

For example, running `astro check --minimumFailingSeverity warning` will cause the command to exit with an error if any warnings are detected.
```

File: src/content/docs/en/reference/cli-reference.mdx [hunk 5]
```diff
@@ -226,6 +226,36 @@ Use these flags to customize the behavior of the command.
...
+
+#### `--minimumSeverity <error|warning|hint>`
+
+Default: `hint`
+
+Minimum severity to output.
+
+For example, choosing `warning` will show errors and warnings, but not hints.
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:251]:
```suggestion
Specifies the minimum severity to output. Defaults to `hint`.

For example, running `astro check --minimumSeverity warning` will show errors and warnings, but not hints.
```

File: src/content/docs/en/reference/cli-reference.mdx [hunk 6]
```diff
@@ -226,6 +226,36 @@ Use these flags to customize the behavior of the command.
...
+
+#### `--preserveWatchOutput`
+
+Default: `false`
+
+If set to false, the output will not be cleared between checks when in watch mode.
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:257]: I'm confused about the wording of this one. If the default is false, then why would someone set this flag to false? Do you mean instead that the default behaviour if you do not include this flag is not to clear output between checks? (Not that you run this flag set to false?)

And, does this actually require being run with a `true` or `false` value? Or is just running with this flag the equivalent of specifying whether or not to clear the output between checks when in watch mode?

Is it more like...?

```suggestion
Specifies not to clear the output between checks when in watch mode.
```

**Review by sarah11918 — COMMENTED** [review]
Thank you so much for participating in this issue, and then coming in to improve the docs @lifeisfakenews ! This is a super helpful way to contribute, and we're so glad you did!

I've edited some wording a bit, to keep it consistent with our structure on the rest of the page. Also, there's one particular flag that needs a bit of clarification to make the definition the most helpful, I think. Can you take a particular look at that one, and seeing the format of the others, help us get that one in a similar state?

**Comment by sarah11918** [comment]
And @at-the-vr thank you for doing a first review tech check on this! It's so helpful! Take a look at the comments I gave on *my* editing pass (checking for consistent wording on other parts of the page etc.) and feel free to edit more anytime you want to up your review game and check for things like that! :raised_hands:

File: src/content/docs/en/reference/cli-reference.mdx (hunk 6 above)
Comment (at-the-vr) [reply src/content/docs/en/reference/cli-reference.mdx:257]: I made this suggestion to @lifeisfakenews simply quoting 
https://github.com/withastro/language-tools/blob/64a46b81f4fb7b2f6c183cb3aebe211629b87bde/packages/astro-check/src/options.ts#L27C11-L31C4
my bad 😖, should have tested before copypasting it.
<hr>

Clarifying the flag:
`astro check --preserveWatchOutput` which is `true` condition preserves the output on the terminal during subsequent `check`.
Not specifying this flag is the `false` or `default` condition which as said clears the output during subsequent `check`
<hr>

providing `true` or `false` as Sarah asked also works smoothly alongside supported flags but it tends to become verbose (I am turning a blind eye to it, leaving it to fellow reviewers 🙌)
<hr>

Suggestion by Sarah is 👍 🔥

**Comment by lifeisfakenews** [comment]
Reworded slightly to follow your suggestions and improved the description for that flag

**Review by sarah11918 — APPROVED** [review]
I think this is looking great! @at-the-vr , would you do one final review on this, and when you're happy, I'm happy!

**Review by at-the-vr — APPROVED** [review]
Lets Git it Merge 💪

File: src/content/docs/en/reference/cli-reference.mdx [hunk 7]
```diff
@@ -226,6 +226,30 @@ Use these flags to customize the behavior of the command.
...
+
+Specifies the minimum severity needed to exit with an error code. Defaults to `error`.
+
+For example, running `astro check --minimumFailingSeverity warning` will cause the command to exit with an error if any warnings are detected.
+
+#### `--minimumSeverity <error|warning|hint>`
+
+Specifies the minimum severity to output. Defaults to `hint`.
+
+For example, running `astro check --minimumFailingSeverity warning` will show errors and warning, but not hints.
```
Comment (sarah11918) [inline src/content/docs/en/reference/cli-reference.mdx:247]:
```suggestion
For example, running `astro check --minimumSeverity warning` will show errors and warning, but not hints.
```

Comment (lifeisfakenews) [reply src/content/docs/en/reference/cli-reference.mdx]: oh whoops i'll fix that

**Pull request merged**